api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -148,7 +89,9 @@ default TableScan select(String... columns) {
    * @return a table scan which can read append data from {@code fromSnapshotId}
    * exclusive and up to {@code toSnapshotId} inclusive
    */
-  TableScan appendsBetween(long fromSnapshotId, long toSnapshotId);
+  default TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
+    throw new UnsupportedOperationException("Incremental scan is not supported");
```

Comment:
We will need to keep this implementation for at least one minor release ?

Reply:
According to the apache project compatibility rules..

Reply:
this change is not strictly required for this change. I am ok to revert it. I added it here as I was thinking about removing the duplicate code of `UnsupportedOperationException` in many `TableScan` impl classes.

We can mark those two `appends` methods as deprecated once the new IncrementalScan impls are ready. Yes, we can follow the compatibility rules.